railties/lib/rails/generators/actions.rb

```diff
@@ -232,6 +232,16 @@ def readme(path)
         log File.read(find_in_source_paths(path))
       end
 
+      # Opens the given file in the user's editor, if generator is run with the <tt>--editor</tt> option.
+      #
+      #   open_file_in_editor "app/models/account.rb"
+      def open_file_in_editor(path)
+        # Attempt to only open the main file generated by the base generator.
+        if shell.base.class === self || shell.base.class.invocations.key?(:orm)
+          run("#{options["editor"]} \"#{path}\"")
```

Comment:
Have you tested with command-line editor like `vim`? Although it opens template, the generator is most likely still running and also outputting to the terminal.